Login is broken on latest dev

When trying to log in on the latest dev, the following error is thrown. 

```
Error during user login
I/flutter ( 4661): type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'
I/flutter ( 4661): #0      AuthenticationServiceCentralized.loginUser (package:junto_beta_mobile/backend/services/auth_service.dart:37:23)
I/flutter ( 4661): <asynchronous suspension>
I/flutter ( 4661): #1      AuthRepo.loginUser (package:junto_beta_mobile/backend/repositories/auth_repo.dart:98:48)
I/flutter ( 4661): #2      AuthBloc._mapLoginEventState (package:junto_beta_mobile/screens/welcome/bloc/auth_bloc.dart:87:35)
I/flutter ( 4661): <asynchronous suspension>
I/flutter ( 4661): #3      AuthBloc.mapEventToState (package:junto_beta_mobile/screens/welcome/bloc/auth_bloc.dart:43:14)
I/flutter ( 4661): <asynchronous suspension>
I/flutter ( 4661): #4      Bloc._bindEventsToStates.<anonymous closure> (package:bloc/src/bloc.dart:252:20)
I/flutter ( 4661): #5      Stream.asyncExpand.<anonymous closure>.<anonymous closure> (dart:async/stream.dart:573:30)
I/flutter ( 4661): #6      _rootRunUnary (dart:async/zone.dart:1206:13)
I/flutter ( 4661): #7      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
I/flutter ( 4661): #8      _CustomZone.runUnaryGuarded (dart:async/zone.dart:1005:7)
I/flutter ( 4661): #9      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:357:11)
I/flutter ( 4661): #10     _DelayedData.perform (dart:async/stream_impl.dart:611:14)
I/flutter ( 4661): #11     _StreamImplEvents.handleNext (dart:async/stream_impl.dart:730:11)
I/flutter ( 4661): #12     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:687:7)
I/flutter ( 4661): #13     _rootRun (dart:async/zone.dart:1182:47)
I/flutter ( 4661): #14     _CustomZone.run (dart:async/zone.dart:1093:19)
I/flutter ( 4661): #15     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
I/flutter ( 4661): #16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
I/flutter ( 4661): #17     _rootRun (dart:async/zone.dart:1190:13)
I/flutter ( 4661): #18     _CustomZone.run (dart:async/zone.dart:1093:19)
I/flutter ( 4661): #19     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
I/flutter ( 4661): #20     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
I/flutter ( 4661): #21     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
I/flutter ( 4661): #22     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
I/flutter ( 4661): 
I/flutter ( 4661): [ERROR] [PrintLogger.logException]: Caught error: type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'
I/flutter ( 4661): [DEBUG] [PrintLogger.logException]: In dev mode. Not sending report to Sentry.io.
I/flutter ( 4661): type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'
I/flutter ( 4661): [ERROR] [PrintLogger.logException]: Caught error: type '_InternalLinkedHashMap<String, dynamic>' is not a subtype of type 'String'
I/flutter ( 4661): [DEBUG] [PrintLogger.logException]: In dev mode. Not sending report to Sentry.io.
```

--- a/.gitignore
+++ b/.gitignore
@@ -93,3 +93,4 @@ ios/Runner/awsconfiguration.json
 android/app/src/main/res/raw/awsconfiguration.json
 android/app/src/tst/res/raw/awsconfiguration.json
 android/app/src/debug/res/raw/awsconfiguration.json
+awsconfiguration.json

--- a/lib/app/app.dart
+++ b/lib/app/app.dart
@@ -24,6 +24,10 @@ class JuntoApp extends StatelessWidget {
   Widget build(BuildContext context) {
     return MultiProvider(
       providers: <SingleChildWidget>[
+        ChangeNotifierProvider<UserDataProvider>(
+          create: (ctx) => backend.dataProvider,
+          lazy: false,
+        ),
         ChangeNotifierProvider<JuntoThemesProvider>.value(
             value: backend.themesProvider),
         Provider<SearchService>.value(value: backend.searchRepo),
@@ -39,13 +43,6 @@ class JuntoApp extends StatelessWidget {
         ChangeNotifierProvider<AppRepo>.value(value: backend.appRepo),
         ChangeNotifierProvider(
           create: (_) => NotificationsHandler(backend.notificationRepo),
-          lazy: false,
-        ),
-        ChangeNotifierProvider<UserDataProvider>(
-          create: (ctx) => UserDataProvider(
-            ctx.repository<AppRepo>(),
-            ctx.repository<UserRepo>(),
-          ),
           lazy: false,
         ),
       ],

--- a/lib/app/app.dart
+++ b/lib/app/app.dart
@@ -1,5 +1,4 @@
 import 'package:flutter/material.dart';
-import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:junto_beta_mobile/app/material_app_with_theme.dart';
 import 'package:junto_beta_mobile/app/providers/bloc_providers.dart';
 import 'package:junto_beta_mobile/app/themes_provider.dart';

--- a/lib/app/app.dart
+++ b/lib/app/app.dart
@@ -23,9 +23,8 @@ class JuntoApp extends StatelessWidget {
   Widget build(BuildContext context) {
     return MultiProvider(
       providers: <SingleChildWidget>[
-        ChangeNotifierProvider<UserDataProvider>(
-          create: (ctx) => backend.dataProvider,
-          lazy: false,
+        ChangeNotifierProvider<UserDataProvider>.value(
+          value: backend.dataProvider,
         ),
         ChangeNotifierProvider<JuntoThemesProvider>.value(
             value: backend.themesProvider),

--- a/lib/app/providers/bloc_providers.dart
+++ b/lib/app/providers/bloc_providers.dart
@@ -1,6 +1,7 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:junto_beta_mobile/backend/backend.dart';
+import 'package:junto_beta_mobile/backend/repositories/onboarding_repo.dart';
 import 'package:junto_beta_mobile/filters/bloc/channel_filtering_bloc.dart';
 import 'package:junto_beta_mobile/models/expression_query_params.dart';
 import 'package:junto_beta_mobile/screens/collective/bloc/collective_bloc.dart';
@@ -21,6 +22,7 @@ class BlocProviders extends StatelessWidget {
             ctx.repository<AuthRepo>(),
             ctx.repository<UserDataProvider>(),
             ctx.repository<UserRepo>(),
+            ctx.repository<OnBoardingRepo>()
           ),
         ),
         BlocProvider<PerspectivesBloc>(

--- a/lib/backend/backend.dart
+++ b/lib/backend/backend.dart
@@ -21,6 +21,7 @@ import 'package:junto_beta_mobile/backend/services/image_handler.dart';
 import 'package:junto_beta_mobile/backend/services/notification_service.dart';
 import 'package:junto_beta_mobile/backend/services/search_service.dart';
 import 'package:junto_beta_mobile/backend/services/user_service.dart';
+import 'package:junto_beta_mobile/backend/user_data_provider.dart';
 import 'package:junto_beta_mobile/utils/junto_http.dart';
 
 export 'package:junto_beta_mobile/backend/repositories.dart';
@@ -40,6 +41,7 @@ class Backend {
     this.db,
     this.themesProvider,
     this.onBoardingRepo,
+    this.dataProvider,
   });
 
   // ignore: missing_return
@@ -68,9 +70,10 @@ class Backend {
       final searchService = SearchServiceCentralized(client);
       final notificationService = NotificationServiceImpl(client);
       final notificationRepo = NotificationRepo(notificationService, dbService);
-
+      final appRepo = AppRepo();
       final userRepo =
           UserRepo(userService, notificationRepo, dbService, expressionService);
+      final dataProvider = UserDataProvider(appRepo, userRepo);
       return Backend._(
         searchRepo: SearchRepo(searchService),
         authRepo: authRepo,
@@ -80,10 +83,11 @@ class Backend {
         expressionRepo:
             ExpressionRepo(expressionService, dbService, imageHandler),
         notificationRepo: notificationRepo,
-        appRepo: AppRepo(),
+        appRepo: appRepo,
         db: dbService,
         themesProvider: themesProvider,
-        onBoardingRepo: OnBoardingRepo(),
+        dataProvider: dataProvider,
+        onBoardingRepo: OnBoardingRepo(dataProvider),
       );
     } catch (e, s) {
       logger.logException(e, s);
@@ -102,11 +106,11 @@ class Backend {
       userRepo: UserRepo(userService, null, null, expressionService),
       collectiveProvider: null,
       groupsProvider: GroupRepo(groupService, userService),
-      //TODO(Nash): MockDB
       expressionRepo: ExpressionRepo(expressionService, null, imageHandler),
       searchRepo: SearchRepo(searchService),
       appRepo: AppRepo(),
       db: null,
+      dataProvider: null,
       themesProvider: MockedThemesProvider(),
       onBoardingRepo: null,
     );
@@ -123,4 +127,5 @@ class Backend {
   final LocalCache db;
   final ThemesProvider themesProvider;
   final OnBoardingRepo onBoardingRepo;
+  final UserDataProvider dataProvider;
 }

--- a/lib/backend/repositories/onboarding_repo.dart
+++ b/lib/backend/repositories/onboarding_repo.dart
@@ -1,9 +1,12 @@
 import 'package:hive/hive.dart';
+import 'package:junto_beta_mobile/app/logger/logger.dart';
+import 'package:junto_beta_mobile/backend/backend.dart';
 import 'package:junto_beta_mobile/hive_keys.dart';
 
 class OnBoardingRepo {
-  OnBoardingRepo() {
-    _loadTutorialState();
+  OnBoardingRepo(UserDataProvider repo) {
+    _repo = repo;
+    loadTutorialState();
   }
   bool _showLotusTutorial = false;
   bool _showPerspectiveTutorial = false;
@@ -11,7 +14,9 @@ class OnBoardingRepo {
   bool _showPackTutorial = false;
   bool _showDenTutorial = false;
   bool _showCreateTutorial = false;
+  bool _showRelationTutorial = false;
   Box _appBox;
+  UserDataProvider _repo;
 
   bool get showLotusTutorial => _showLotusTutorial;
   bool get showPerspectiveTutorial => _showPerspectiveTutorial;
@@ -19,49 +24,59 @@ class OnBoardingRepo {
   bool get showPackTutorial => _showPackTutorial;
   bool get showDenTutorial => _showDenTutorial;
   bool get showCreateTutorial => _showCreateTutorial;
+  bool get showRelationTutorial => _showRelationTutorial;
 
-  Future<void> _loadTutorialState() async {
-    return;
-//    _appBox = await Hive.box(HiveBoxes.kAppBox);
-//    _showLotusTutorial = await _appBox.get(HiveKeys.kShowLotusTutorial) ?? true;
-//    _showCollectiveTutorial =
-//        await _appBox.get(HiveKeys.kShowCollectiveTutorial) ?? true;
-//    _showPackTutorial = await _appBox.get(HiveKeys.kShowPackTutorial) ?? true;
-//    _showDenTutorial = await _appBox.get(HiveKeys.kShowDenTutorial) ?? true;
-//    _showPerspectiveTutorial =
-//        await _appBox.get(HiveKeys.kShowPerspectiveTutorial) ?? true;
-//    _showCreateTutorial =
-//        await _appBox.get(HiveKeys.kShowCreateTutorial) ?? true;
+  Future<void> loadTutorialState() async {
+    final time = DateTime.now();
+    final timeDifference =
+        _repo.userProfile?.pack?.createdAt?.difference(time)?.inMinutes;
+    if (_repo.userProfile != null && timeDifference < 1) {
+      _appBox = await Hive.box(HiveBoxes.kAppBox);
+      _showLotusTutorial =
+          await _appBox.get(HiveKeys.kShowLotusTutorial) ?? true;
+      _showCollectiveTutorial =
+          await _appBox.get(HiveKeys.kShowCollectiveTutorial) ?? true;
+      _showPackTutorial = await _appBox.get(HiveKeys.kShowPackTutorial) ?? true;
+      _showDenTutorial = await _appBox.get(HiveKeys.kShowDenTutorial) ?? true;
+      _showPerspectiveTutorial =
+          await _appBox.get(HiveKeys.kShowPerspectiveTutorial) ?? true;
+      _showCreateTutorial =
+          await _appBox.get(HiveKeys.kShowCreateTutorial) ?? true;
+      _showRelationTutorial =
+          await _appBox.get(HiveKeys.kRelationsTutorial) ?? true;
+    }
   }
 
   Future<void> setViewed(String key, bool value) async {
-    //TODO: Re-enable once api is updated
-    return;
-//    switch (key) {
-//      case HiveKeys.kShowLotusTutorial:
-//        _showLotusTutorial = false;
-//        await _appBox.put(HiveKeys.kShowLotusTutorial, value);
-//        return;
-//      case HiveKeys.kShowCollectiveTutorial:
-//        _showCollectiveTutorial = false;
-//        await _appBox.put(HiveKeys.kShowCollectiveTutorial, value);
-//        return;
-//      case HiveKeys.kShowPackTutorial:
-//        _showPackTutorial = false;
-//        await _appBox.put(HiveKeys.kShowPackTutorial, value);
-//        return;
-//      case HiveKeys.kShowDenTutorial:
-//        _showDenTutorial = false;
-//        await _appBox.put(HiveKeys.kShowDenTutorial, value);
-//        return;
-//      case HiveKeys.kShowPerspectiveTutorial:
-//        _showPerspectiveTutorial = false;
-//        await _appBox.put(HiveKeys.kShowPerspectiveTutorial, value);
-//        return;
-//      case HiveKeys.kShowCreateTutorial:
-//        _showCreateTutorial = false;
-//        await _appBox.put(HiveKeys.kShowCreateTutorial, value);
-//        return;
-//    }
+    switch (key) {
+      case HiveKeys.kShowLotusTutorial:
+        _showLotusTutorial = false;
+        await _appBox.put(HiveKeys.kShowLotusTutorial, value);
+        return;
+      case HiveKeys.kShowCollectiveTutorial:
+        _showCollectiveTutorial = false;
+        await _appBox.put(HiveKeys.kShowCollectiveTutorial, value);
+        return;
+      case HiveKeys.kShowPackTutorial:
+        _showPackTutorial = false;
+        await _appBox.put(HiveKeys.kShowPackTutorial, value);
+        return;
+      case HiveKeys.kShowDenTutorial:
+        _showDenTutorial = false;
+        await _appBox.put(HiveKeys.kShowDenTutorial, value);
+        return;
+      case HiveKeys.kShowPerspectiveTutorial:
+        _showPerspectiveTutorial = false;
+        await _appBox.put(HiveKeys.kShowPerspectiveTutorial, value);
+        return;
+      case HiveKeys.kShowCreateTutorial:
+        _showCreateTutorial = false;
+        await _appBox.put(HiveKeys.kShowCreateTutorial, value);
+        return;
+      case HiveKeys.kRelationsTutorial:
+        _showRelationTutorial = false;
+        await _appBox.put(HiveKeys.kRelationsTutorial, value);
+        return;
+    }
   }
 }

--- a/lib/backend/repositories/onboarding_repo.dart
+++ b/lib/backend/repositories/onboarding_repo.dart
@@ -1,5 +1,4 @@
 import 'package:hive/hive.dart';
-import 'package:junto_beta_mobile/app/logger/logger.dart';
 import 'package:junto_beta_mobile/backend/backend.dart';
 import 'package:junto_beta_mobile/hive_keys.dart';
 

--- a/lib/hive_keys.dart
+++ b/lib/hive_keys.dart
@@ -1,6 +1,7 @@
 /// Various item keys used to store data in Hive boxes
 class HiveKeys {
   /// Used to access
+  static const kisLoggedIn = 'isLoggedIn';
   static const kUserFollowPerspectiveId = 'userFollowPerspectiveId';
   static const kUserId = 'userId';
   static const kUserData = 'user_data';
@@ -15,6 +16,7 @@ class HiveKeys {
   static const kShowDenTutorial = "show-den-tutorial";
   static const kShowPerspectiveTutorial = "show-perspective-tutorial";
   static const kShowCreateTutorial = "show-create-tutorial";
+  static const kRelationsTutorial = "show-relation-tutorial";
 }
 
 /// Class containing the name of the different "boxes" used by the application.

--- a/lib/models/auth_result.dart
+++ b/lib/models/auth_result.dart
@@ -1,5 +1,4 @@
 import 'package:flutter/foundation.dart';
-import 'package:flutter_aws_amplify_cognito/flutter_aws_amplify_cognito.dart';
 
 abstract class AuthResult {
   final bool wasSuccessful;

--- a/lib/screens/collective/collective.dart
+++ b/lib/screens/collective/collective.dart
@@ -3,7 +3,6 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:junto_beta_mobile/backend/repositories/app_repo.dart';
-import 'package:junto_beta_mobile/models/perspective.dart';
 import 'package:junto_beta_mobile/models/expression_query_params.dart';
 import 'package:junto_beta_mobile/screens/collective/collective_fab.dart';
 import 'package:junto_beta_mobile/screens/collective/perspectives/bloc/perspectives_bloc.dart';

--- a/lib/screens/collective/perspectives/about_perspective.dart
+++ b/lib/screens/collective/perspectives/about_perspective.dart
@@ -212,9 +212,9 @@ class AboutPerspectiveMembers extends StatelessWidget {
     if (perspective.name == 'JUNTO' && perspective.isDefault == true) {
       return 'All Members';
     } else if (perspective.userCount == 1) {
-      return perspective.userCount.toString() + ' Member';
+      return '${perspective.userCount} Member';
     } else {
-      return perspective.userCount.toString() + ' Members';
+      return '${perspective.userCount} Members';
     }
   }
 

--- a/lib/screens/collective/perspectives/collective_populated_list.dart
+++ b/lib/screens/collective/perspectives/collective_populated_list.dart
@@ -1,5 +1,4 @@
 import 'package:flutter/material.dart';
-import 'package:junto_beta_mobile/backend/backend.dart';
 import 'package:junto_beta_mobile/backend/repositories/app_repo.dart';
 import 'package:junto_beta_mobile/models/models.dart';
 import 'package:junto_beta_mobile/screens/collective/bloc/collective_bloc.dart';

--- a/lib/screens/create/create_actions/create_actions.dart
+++ b/lib/screens/create/create_actions/create_actions.dart
@@ -13,7 +13,6 @@ import 'package:junto_beta_mobile/screens/collective/collective.dart';
 import 'package:junto_beta_mobile/screens/create/create_actions/channel_search_modal.dart';
 import 'package:junto_beta_mobile/screens/create/create_actions/create_actions_appbar.dart';
 import 'package:junto_beta_mobile/screens/packs/packs.dart';
-import 'package:junto_beta_mobile/screens/packs/packs_bloc/pack_bloc.dart';
 import 'package:junto_beta_mobile/utils/junto_overlay.dart';
 import 'package:junto_beta_mobile/utils/utils.dart';
 import 'package:junto_beta_mobile/widgets/dialogs/single_action_dialog.dart';

--- a/lib/screens/create/create_actions/widgets/create_app_bar.dart
+++ b/lib/screens/create/create_actions/widgets/create_app_bar.dart
@@ -1,7 +1,6 @@
 import 'package:flutter/material.dart';
 import 'package:junto_beta_mobile/app/expressions.dart';
 import 'package:junto_beta_mobile/screens/create/create_actions/widgets/cta_button.dart';
-import 'package:junto_beta_mobile/app/custom_icons.dart';
 
 class CreateAppBar extends StatelessWidget implements PreferredSizeWidget {
   const CreateAppBar({

--- a/lib/screens/lotus/lotus.dart
+++ b/lib/screens/lotus/lotus.dart
@@ -160,7 +160,8 @@ class _JuntoLotusState extends State<JuntoLotus> {
                                         featureId: 'lotus_info_id',
                                         oneFeature: true,
                                         title:
-                                            "Welcome to Junto! You're looking at the app's tutorial, which you can open and close by clicking the question mark on the top right corner of your screen. To begin, this is the intention screen, your medium for navigation.",
+                                            "Welcome to Junto! You're looking at the app's tutorial, which you can open and close by clicking the question mark on the top right corner of your screen.  "
+                                            " To begin, this is the intention screen, your medium for navigation.",
                                         learnMore: true,
                                         learnMoreText: [
                                           'The purpose of the intention screen is to reduce noise and to invite a more mindful, self-directed experience. Instead of immediately being dropped into a feed when you open the app, this screen makes your initial experience a choice and encourages more reflection while navigating between screens.'

--- a/lib/screens/packs/pack_open/pack_open_action_items.dart
+++ b/lib/screens/packs/pack_open/pack_open_action_items.dart
@@ -85,7 +85,7 @@ class _PackOpenActionItemsState extends State<PackOpenActionItems> {
                   title: Row(
                     children: <Widget>[
                       Text(
-                        'View @' + widget.pack.creator['username'] + "'s den",
+                        "'View @' + ${widget.pack.creator['username']}'s den",
                         style: Theme.of(context).textTheme.headline5,
                       ),
                     ],

--- a/lib/screens/packs/pack_open/pack_open_action_items.dart
+++ b/lib/screens/packs/pack_open/pack_open_action_items.dart
@@ -26,6 +26,22 @@ class _PackOpenActionItemsState extends State<PackOpenActionItems> {
     Navigator.pop(context);
   }
 
+  Future<void> _navigateMember() async {
+    final String packCreatorAddress = widget.pack.creator['address'];
+    Navigator.pop(context);
+    final UserData userData =
+        await Provider.of<UserRepo>(context, listen: false)
+            .getUser(packCreatorAddress);
+
+    final UserProfile _userProfile = userData.user;
+    Navigator.push(
+      context,
+      CupertinoPageRoute(
+        builder: (context) => JuntoMember(profile: _userProfile),
+      ),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
     return Container(
@@ -65,23 +81,7 @@ class _PackOpenActionItemsState extends State<PackOpenActionItems> {
                 const SizedBox(height: 10),
                 ListTile(
                   contentPadding: const EdgeInsets.all(0),
-                  onTap: () async {
-                    final String packCreatorAddress =
-                        widget.pack.creator['address'];
-                    Navigator.pop(context);
-                    final UserData userData =
-                        await Provider.of<UserRepo>(context, listen: false)
-                            .getUser(packCreatorAddress);
-
-                    final UserProfile _userProfile = userData.user;
-                    Navigator.push(
-                      context,
-                      CupertinoPageRoute(
-                        builder: (context) =>
-                            JuntoMember(profile: _userProfile),
-                      ),
-                    );
-                  },
+                  onTap: _navigateMember,
                   title: Row(
                     children: <Widget>[
                       Text(

--- a/lib/screens/welcome/bloc/auth_bloc.dart
+++ b/lib/screens/welcome/bloc/auth_bloc.dart
@@ -3,6 +3,7 @@ import 'dart:async';
 import 'package:bloc/bloc.dart';
 import 'package:junto_beta_mobile/app/logger/logger.dart';
 import 'package:junto_beta_mobile/backend/backend.dart';
+import 'package:junto_beta_mobile/backend/repositories/onboarding_repo.dart';
 import 'package:junto_beta_mobile/utils/junto_exception.dart';
 
 import 'bloc.dart';
@@ -11,8 +12,14 @@ class AuthBloc extends Bloc<AuthEvent, AuthState> {
   final AuthRepo authRepo;
   final UserRepo userRepo;
   final UserDataProvider userDataProvider;
+  final OnBoardingRepo onBoardingRepo;
 
-  AuthBloc(this.authRepo, this.userDataProvider, this.userRepo) {
+  AuthBloc(
+    this.authRepo,
+    this.userDataProvider,
+    this.userRepo,
+    this.onBoardingRepo,
+  ) {
     _getLoggedIn();
   }
 
@@ -64,7 +71,7 @@ class AuthBloc extends Bloc<AuthEvent, AuthState> {
       }
       await userDataProvider.updateUser(userData);
       await userDataProvider.initialize();
-
+      await onBoardingRepo.loadTutorialState();
       yield AuthState.agreementsRequired(userData);
     } on JuntoException catch (e) {
       logger.logError('Error during sign up: ${e.message}');

--- a/lib/widgets/end_drawer/end_drawer.dart
+++ b/lib/widgets/end_drawer/end_drawer.dart
@@ -14,7 +14,6 @@ import 'package:junto_beta_mobile/widgets/avatars/member_avatar.dart';
 import 'package:junto_beta_mobile/widgets/background/background_theme.dart';
 import 'package:junto_beta_mobile/widgets/dialogs/confirm_dialog.dart';
 import 'package:junto_beta_mobile/widgets/end_drawer/end_drawer_relationships/end_drawer_relationships.dart';
-import 'package:junto_beta_mobile/widgets/end_drawer/junto_resources.dart';
 import 'package:junto_beta_mobile/widgets/end_drawer/junto_account.dart';
 import 'package:junto_beta_mobile/widgets/fade_route.dart';
 import 'package:junto_beta_mobile/widgets/utils/app_version_label.dart';

--- a/lib/widgets/end_drawer/end_drawer_relationships/end_drawer_relationships.dart
+++ b/lib/widgets/end_drawer/end_drawer_relationships/end_drawer_relationships.dart
@@ -4,6 +4,8 @@ import 'package:flutter/material.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:junto_beta_mobile/app/custom_icons.dart';
 import 'package:junto_beta_mobile/backend/repositories.dart';
+import 'package:junto_beta_mobile/backend/repositories/onboarding_repo.dart';
+import 'package:junto_beta_mobile/hive_keys.dart';
 import 'package:junto_beta_mobile/models/models.dart';
 import 'package:junto_beta_mobile/widgets/notification_signal.dart';
 import 'package:junto_beta_mobile/widgets/tab_bar.dart';
@@ -55,7 +57,24 @@ class JuntoRelationshipsState extends State<JuntoRelationships> {
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
+    final repo = Provider.of<OnBoardingRepo>(context);
+    if (repo.showRelationTutorial) {
+      showTutorial();
+      repo.setViewed(HiveKeys.kRelationsTutorial, false);
+    }
+  }
+
+  void showTutorial() {
     getUserRelationships();
+    FeatureDiscovery.clearPreferences(context, <String>{
+      'relations_info_id',
+    });
+    FeatureDiscovery.discoverFeatures(
+      context,
+      const <String>{
+        'relations_info_id',
+      },
+    );
   }
 
   @override

--- a/lib/widgets/previews/expression_preview/two_column_preview/two_column_expression_preview.dart
+++ b/lib/widgets/previews/expression_preview/two_column_preview/two_column_expression_preview.dart
@@ -4,7 +4,6 @@ import 'package:junto_beta_mobile/backend/backend.dart';
 import 'package:junto_beta_mobile/models/expression.dart';
 import 'package:junto_beta_mobile/screens/expression_open/expression_open.dart';
 import 'package:junto_beta_mobile/utils/utils.dart';
-import 'package:junto_beta_mobile/widgets/fade_route.dart';
 import 'package:junto_beta_mobile/widgets/previews/expression_preview/expression_preview_bottom.dart';
 import 'package:junto_beta_mobile/widgets/previews/expression_preview/two_column_preview/two_column_expression_preview_types/dynamic.dart';
 import 'package:junto_beta_mobile/widgets/previews/expression_preview/two_column_preview/two_column_expression_preview_types/event.dart';

